### PR TITLE
Add some additional metrics for Minion tasks

### PR DESCRIPTION
**Pull request by vvivekiyer**
PR adds some additional metrics for Minion tasks. Adds the following metrics:

  SEGMENT_DOWNLOAD_FAIL_COUNT
  SEGMENT_DOWNLOAD_COUNT
  SEGMENT_UPLOAD_COUNT
  SEGMENT_BYTES_DOWNLOADED
  SEGMENT_BYTES_UPLOADED
  RECORDS_PROCESSED_COUNT
  RECORDS_PURGED_COUNT

### CODE REVIEW

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/MinionMeter.java [hunk 1]
```diff
@@ -31,7 +31,13 @@ public enum MinionMeter implements AbstractMetrics.Meter {
   NUMBER_TASKS_FAILED("tasks", false),
   NUMBER_TASKS_FATAL_FAILED("tasks", false),
   SEGMENT_UPLOAD_FAIL_COUNT("segments", false),
-  SEGMENT_DOWNLOAD_FAIL_COUNT("segments", false);
+  SEGMENT_DOWNLOAD_FAIL_COUNT("segments", false),
+  SEGMENTS_DOWNLOADED("segments", false),
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/MinionMeter.java:35]: Can we try to follow one convention for the metric name?

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/MinionMeter.java [hunk 2]
```diff
@@ -31,7 +31,13 @@ public enum MinionMeter implements AbstractMetrics.Meter {
...
+  SEGMENTS_UPLOADED("segments", false),
+  SEGMENT_SIZE_DOWNLOADED("bytes", false),
+  SEGMENT_SIZE_UPLOADED("bytes", false),
+  RECORDS_PER_SEGMENT("rows", false),
```
Comment (Jackie-Jiang) [inline pinot-common/src/main/java/org/apache/pinot/common/metrics/MinionMeter.java:39]: What does `per` segment stand for? I don't see we are counting the average

Comment (sajjad-moradi) [reply pinot-common/src/main/java/org/apache/pinot/common/metrics/MinionMeter.java:39]: +1

File: pinot-common/src/main/java/org/apache/pinot/common/metrics/MinionMeter.java (hunk 1 above)
Comment (vvivekiyer) [reply pinot-common/src/main/java/org/apache/pinot/common/metrics/MinionMeter.java]: Renamed.

**Review by sajjad-moradi — APPROVED** [review]

**Pull request merged**